src/Array.ts

```diff
@@ -0,0 +1,406 @@
+ */
+export function partialEquals<T>(array1: T[], offset1: number, array2: T[], offset2: number, length: number): boolean;
+/**
+ * Compare two part of arrays to check them equality.
+ *
+ * 2つの配列の一部が等しいか比較します。
+ * @param array1 array
+ * @param offset1 beginning index on array1
+ * @param array2 array
+ * @param offset2 beginning index on array2
+ * @param length length to compare
+ * @param equal equality comparing function
+ * @returns partial equality
+ */
+export function partialEquals<T>(
+    array1: T[], offset1: number, array2: T[], offset2: number, length: number,
+    equal: (v1: T, v2: T) => boolean
+): boolean;
+
+export function partialEquals<T>(array1: T[], ...args: any[]): boolean {
+    let offset1: number = 0;
+    let array2: T[];
+    let offset2: number = 0;
+    let len: number;
+
+    switch (args.length) {
+    case 2:
+        [array2, len] = args;
+        break;
+    case 4:
+        [offset1, array2, offset2, len] = args;
+        break;
+    case 5:
+        let eq: (v1: T, v2: T) => boolean;
+        [offset1, array2, offset2, len, eq] = args;
+        return _partialEquals(array1, offset1, array2, offset2, len, eq);
+    default:
+        throw new Error("Invalid overload arguments");
+    }
+
+    return _defaultPartialEquals(array1, offset1, array2, offset2, len);
+}
+
+function _defaultPartialEquals<T>(
+    array1: T[], offset1: number, array2: T[], offset2: number, length: number,
+): boolean {
+    for (let i = 0; i < length; i++) {
+        if (array1[offset1 + i] !== array2[offset2 + i]) {
+            return false;
+        }
+    }
+    return true;
+}
+
+function _partialEquals<T>(
+    array1: T[], offset1: number, array2: T[], offset2: number, length: number,
+    eq: (v1: T, v2: T) => boolean
+) {
+    for (let i = 0; i < length; i++) {
+        if (!eq(array1[offset1 + i], array2[offset2 + i])) {
+            return false;
+        }
+    }
+    return true;
+}
+
+
+export function startsWith<T>(array: T[], prefix: T[]): boolean {
+    if (!(prefix?.length <= array?.length)) {
+        return false;
+    }
+
+    for (let i = 0; i < prefix.length; i++) {
+        if (array[i] !== prefix[i]) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+
+export function endsWith<T>(array: T[], suffix: T[]): boolean {
+    const offset = array?.length - suffix?.length;
+    if (!(0 <= offset)) {
+        return false;
+    }
+
+    for (let i = 0; i < suffix.length; i++) {
+        if (array[offset + i] !== suffix[i]) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+
+/**
+ * Split array to two arrays.
+ *
+ * 配列を2つに分割します。
+ * @param array array
+ * @param length length of first array
+ * @returns split arrays
+ */
+export function split<T>(array: T[], length: number): [T[], T[]] {
+    let len = 0 <= length ? length : array.length + length;
+    len = clip(len, 0, array.length);
+    return [array.slice(0, len), array.slice(len)];
+}
+
+
+/**
+ * 
```

Comment:
Codacy has a fix for the issue: [Trailing spaces not allowed.](https://app.codacy.com/gh/simpler-one/node.utils/pullRequest?prid=7629571)
```suggestion
 *
```